File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        Stack<Character> parenthesis = new Stack<>();
        int size = s.length() - 1;
        for (int i = 0; i <= size; i++)
        {
            char ch = s.charAt(i);
            if(parenthesis.empty())
                parenthesis.push(ch);
            else if(ch == ')' && parenthesis.peek() == '('){
                parenthesis.pop();
            } 
            else parenthesis.push(ch);
        }
        return parenthesis.size();
    }
}

File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        stack<char> parenthesis;
        for (int i = 0; i < s.size(); i++)
        {
            if(parenthesis.empty())
               parenthesis.push(s[i]);
            else if(s[i] == ')' && parenthesis.top() == '('){
                parenthesis.pop();
            } 
            else parenthesis.push(s[i]);
        }
        return parenthesis.size();
    }
};